src/main/java/frc/robot/subsystems/Feeder.java

```diff
@@ -31,12 +53,48 @@ public void stopFeederMotor() {
     public boolean getFeederSensor() {
         return feederSensor.get();
     }
+    public Color getColor() {
+        return m_colorSensor.getColor();
+    }
+    public boolean pieceSeen() {
+        return (piece == ColorObject.Piece);
+    }
 
     @Override
     public void periodic() {
-        if (Constants.FEEDER_DEBUG){
+        Color detectedColor = m_colorSensor.getColor();
+        double IR = m_colorSensor.getIR();
+        String colorObject = "";
+        double proximity = m_colorSensor.getProximity();
+
+        ColorMatchResult matchedColor = m_colorMatcher.matchClosestColor(detectedColor);
+        if (matchedColor.color == GameConstants.C_PIECE_TARGET)
+            piece = ColorObject.Piece;
+        else if (matchedColor.color == GameConstants.C_PLASTIC_TARGET)
+            piece = ColorObject.Plastic;
+        else
+            piece = ColorObject.None;
+        switch (piece) {
+            case Piece:
+                colorObject = "Piece";
+                break;
+            case Plastic:
+                colorObject = "Plastic";
+                break;
+            case None:
+                colorObject = "None";
```

Comment:
You can add the name of the color object to the enum and eliminate this switch statement.